openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -98,6 +98,7 @@ def __init__(self, course_key, request, username=''):
             course_key=course_key,
             is_global_staff=self.original_user_is_global_staff,
         )
+        self.can_view_certificate = certificates_viewable_for_course(self.course)
```

Comment:
Nit (This is all optional and are just some thoughts): It looks like `certificates_viewable_for_course` expects a CourseOverview, but the `self.course` is the actual course object from the modulestore, not a CourseOverview.

Perhaps we should retrieve the course overview and store it as part of the `CoursewareMeta` object? Then we can pass the actual course overview to `certificates_viewable_for_course` and not rely on the logic within that method to retrieve the course overview for us (which seems weird?).

If we do this now since we're already making changes, maybe this makes it easier in the future for us to remove this use of the modulestore to retrieve course info.

Reply:
Ahh okay, I like that idea. I'll work on making the call to CourseOverview instead.